mapof.go

```diff
@@ -140,7 +150,7 @@ func (m *MapOf[V]) doStore(key string, value V, loadIfExists bool) (V, bool) {
 			if !topHashMatch(hash, b.topHashes, i) {
 				continue
 			}
-			if key == derefKey(b.keys[i]) {
+			if hash == m.hasher(derefTypedValue[K](b.keys[i])) {
```

Comment:
Equal hash codes don't imply same key. `K` type should be `comparable` to allow equality checks.

Reply:
Addressed in subsequent commit :)